### DISCUSSION TITLE
Prostate multimodality registration

### DISCUSSION THREAD

**Discussion by jakubMitura14** [voxelmorph/voxelmorph#447] [General]
Hello i am working on some prostate multimodal mri dataset with some metadata like age. I have a plan for registration but i am not sure it is optimal 
1) pad all images to size of the biggest in a series as i seen in example algorithm expect fixed sized inputs
2) supply age and list of files from single modality so separately t2 saggital dwi transverse, t2 saggital etc...
3) transform all images to their respective conditional templates
 4)using your model trained on abstract synth morph register modality to modality more precisely  register all in respect to t2w

Problems

Is padding good solution - as low resolution prostates will occupy just the central part of image 

When i have multiple images of t2 in transverse saggital and coronal i should create separate templates for each or just one for all t2w?

Images has diffrent extents - how to deal with it ?

I need also to modify labels that is annotated on t2w i would like to apply the same transformation as is done from t2w to template to label is it possible ?

Should i first register template to template from diffrent modalities?

I am aware that frequently there is no single answer, just a matter of experimentation, yet you know this tool best so i would like to have your opinion .

Thank you for your time reading it !

**Comment by mu40** [comment]
Hi @jakubMitura14. For preprocessing, I would recommend resampling all images at a common isotropic resolution before zero-padding (and/or cropping) to a common shape. This will reduce the amount of zero-padding needed for lower-resolution images, and it will also preserve the size relationship of the anatomy of interest across acquisitions.

Whether creating a single or a separate template for each of the standard sagittal, coronal, and axial views is better is an interesting question. You will probably want to explore both options and compare. While it could be that training with a single orientation leads to more specialized and representative templates, we frequently see that exposing models to greater variability at training increases robustness and accuracy. Resampling your images at an isotropic resolution as suggested above should help when training a single template across views.

You can apply the same transforms you use for your images to the corresponding label maps, too. However, when you transform label maps for preprocessing or evaluation, you will probably want to use nearest-neighbor instead of higher-order interpolation to avoid mixing labels. I hope this helps.

**Reply by jakubMitura14** [reply]
Thanks!
